doc/user_mgmt/user_mgmt_hld.md

```diff
@@ -0,0 +1,595 @@
+The goal is to standardize and define the User management configurations and functionality.<br/>
+It will define the default roles (capabilities), default users and describe the supported configurations
+for users.<br/>
+
+Note:
+The compilition of the feature code will be controlled by a build flag and it will be disabled by default.
+
+###  1.5. <a name='Requirements'></a>Requirements
+####  1.5.1. <a name='Definedefaults'></a>Define defaults
+* Default roles: admin and monitor<br/>
+* Default users: admin and monitor
+####  1.5.2. <a name='Configurations'></a>Configurations
+* **add/delete user**
+
+	Add/Delete a user from system. Default users shouldn't be deleted.<br/>
+    Defining the password is mandatory when adding a new user.
+* **enable/disable user**
+
+	Control login of the user by enabling/disabling user state.
+* **change full-name**
+
+	Change Gecos field (Description) of user
+* **change role (capability)**
+
+	Control the capability of user by changing the role
+* **change password**
+
+	Change the password of a user.
+
+##  2 <a name='Design'></a>Design
+
+###  2.1 <a name='ArchitectureDesign'></a>Architecture Design
+![Arch_diagram](Arch_diagram.png)
+
+(flow description in the chapter below)
+
+###  2.2 <a name='High-LevelDesign'></a>High-Level Design
+
+Sonic has two main users:
+* Admin: Administrator with limited access rights.It is part of sudo group.<br/>
+* Root: Superuser with highest access rights. Login is disabled for this user.
+
+We will define two roles for users:<br/>
+* **admin**: a role for administrator users with write and read permissions.<br/>
+	Each admin user will be part of the following groups:<br/>
+	**primary group**: admin<br/>
+	**secondary groups**: sudo,docker,redis,adm
+
+* **monitor**: a role for monitoring users with read permissions only. <br/>
+	Each monitor user will be part of the following groups:<br/>
+	**primary group**: adm<br/>
+	**secondary groups**: N/A
+
+In addition, we will define two default users:
+* **admin**: User role is admin.<br/>
+	The only change from the current admin in Sonic is the additional group: adm
+
+* **monitor**: User role is monitor. <br/>
+	User with monitoring permissions only .<br/>
+
+**Note**:
+ Group adm is used for system monitoring tasks. Members of this group can read the files under "/var/log"
+
+
+###  2.3 <a name='UsercfdRole'></a>Usercfgd Role:
+
+Usercfgd is the daemon that get events from USER_TABLE in CONF_DB and apply the new configuration on the system.<br/>
+The new daemon should start running after "config-setup.service" (i.e. after database is ready) but before systemd enables "system-logind.service" (i.e. before user can login to system)<br/>
+
+
+![services_diagram](services_diagram.png))<br/>
+
+
+##  3 <a name='Configurationandmanagement'></a>Configuration and management
+
+###  3.1 <a name='ConfigDBTables'></a>ConfigDB Tables
```

Comment:
Using CONFIG_DB seems like a security issue.  CONFIG_DB is exported into "show techsupport".   Superuser/su is not required for "show techsupport".  This will allow non-superuser to export hashed passwords.  Wouldn't it be better to utilize a new keyspace and add support for exporting/importing like config save/import?

Reply:
I agree. 
I am working with the team on a new design for storing users passwords.
Can you please elaborate about the keyspace ? where I can learn more about it ?
I would appreciate if you could provide an example

Reply:
See /var/run/redis/sonic-db/database_config.json on any running switch (keyspace = DB).  Maybe an unused keyspace/DB could be used and not exported during a "show techsupport".  Here is a .json from a switch running 202111 SONiC:
{
    "INSTANCES": {
        "redis": {
            "hostname": "127.0.0.1",
            "port": 6379,
            "unix_socket_path": "/var/run/redis/redis.sock",
            "persistence_for_warm_boot": "yes"
        }
    },
    "DATABASES": {
        "APPL_DB": {
            "id": 0,
            "separator": ":",
            "instance": "redis"
        },
        "ASIC_DB": {
            "id": 1,
            "separator": ":",
            "instance": "redis"
        },
        "COUNTERS_DB": {
            "id": 2,
            "separator": ":",
            "instance": "redis"
        },
        "LOGLEVEL_DB": {
            "id": 3,
            "separator": ":",
            "instance": "redis"
        },
        "CONFIG_DB": {
            "id": 4,
            "separator": "|",
            "instance": "redis"
        },
        "PFC_WD_DB": {
            "id": 5,
            "separator": ":",
            "instance": "redis"
        },
        "FLEX_COUNTER_DB": {
            "id": 5,
            "separator": ":",
            "instance": "redis"
        },
        "STATE_DB": {
            "id": 6,
            "separator": "|",
            "instance": "redis"
        },
        "SNMP_OVERLAY_DB": {
            "id": 7,
            "separator": "|",
            "instance": "redis"
        },
        "RESTAPI_DB": {
            "id": 8,
            "separator": "|",
            "instance": "redis"
        },
        "GB_ASIC_DB": {
            "id": 9,
            "separator": ":",
            "instance": "redis"
        },
        "GB_COUNTERS_DB": {
            "id": 10,
            "separator": ":",
            "instance": "redis"
        },
        "GB_FLEX_COUNTER_DB": {
            "id": 11,
            "separator": ":",
            "instance": "redis"
        },
        "APPL_STATE_DB": {
            "id": 14,
            "separator": ":",
            "instance": "redis"
        }
    },
    "VERSION": "1.0"
}